runtime/src/chplsys.c

```diff
@@ -553,12 +554,64 @@ static void getCpuInfo(int* p_numPhysCpus, int* p_numLogCpus) {
 
   fclose(f);
 
-  if (cpuCores == 0)
-    cpuCores = 1;
-  if (siblings == 0)
-    siblings = 1;
-  if ((*p_numPhysCpus = procs / (siblings / cpuCores)) <= 0)
-    *p_numPhysCpus = 1;
+  if (cpuCores == 0 && siblings == 0) {
+    // We have a limited-format /proc/cpuinfo.
+    // See if the /sys filesystem has any more information for us.
```

Comment:
@gbtitus - The map is insufficiently documented on kernel.org, the most authoritative source, but it turns out that hwloc uses this file too, and parses it as a hex bitmap.  Also, the lscpu command parses it as a hex bitmap.  There is an IBM document that claims it can also be binary.  In that case, hwloc and lscpu will break but the code here will still work.

The top of the function already contains a warning that homogeneity is assumed.

I've put together a branch with additional comments.  I'm not sure they add much value, but if you prefer, I can make this into a PR and merge it.

https://github.com/chapel-lang/chapel/compare/master...dmk42:corecount2

Reply:
@dmk42 - Oh, sorry, I didn't look far enough into the context to realize we'd already commented that this function assumed homogeneity.  Looking at your branch, I think it'd be worthwhile to merge the 2nd comment (about the file's limited documentation and format), but the 1st can be skipped.  I don't need to review the PR if you choose to do this.